exchanges/kraken/kraken_websocket.go

```diff
@@ -927,22 +941,36 @@ func (k *Kraken) wsProcessOrderBookPartial(channelData *WebsocketChannelData, as
 		if len(bids) < 3 {
 			return errors.New("unexpected bids length")
 		}
-		price, err := strconv.ParseFloat(bids[0].(string), 64)
+		priceStr, ok := bids[0].(string)
+		if !ok {
+			return common.GetTypeAssertError("string", bids[0], "price")
+		}
+		price, err := strconv.ParseFloat(priceStr, 64)
 		if err != nil {
 			return err
 		}
-		amount, err := strconv.ParseFloat(bids[1].(string), 64)
+		amountStr, ok := bids[1].(string)
+		if !ok {
+			return common.GetTypeAssertError("string", bids[1], "amount")
+		}
+		amount, err := strconv.ParseFloat(amountStr, 64)
 		if err != nil {
 			return err
 		}
-		timeData, err := strconv.ParseFloat(bids[2].(string), 64)
+		tdStr, ok := bids[2].(string)
+		if !ok {
+			return common.GetTypeAssertError("string", bids[2], "time")
+		}
+		timeData, err := strconv.ParseFloat(tdStr, 64)
 		if err != nil {
 			return err
 		}
 
 		base.Bids[i] = orderbook.Item{
-			Amount: amount,
-			Price:  price,
+			Amount:    amount,
+			StrAmount: amountStr,
```

Comment:
The only thing I can think of is that perhaps it would be good to have an error - likely at LoadSnapshot()/Update() level - to comment specifically on if the Str versions are populated and the Float versions are not. We still require the float versions to process things and output certain results. I suppose you could do the conversions further down the line if float not populated, but I think that becomes complicated.